example-site/app/Main.hs

```diff
@@ -1,130 +1,121 @@
-{-# LANGUAGE DeriveGeneric #-}
-{-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE OverloadedStrings #-}
-{-# LANGUAGE NamedFieldPuns #-}
-{-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns        #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
 
 module Main where
 
-import Control.Lens
-import Data.Aeson as A
-import Data.Aeson.Lens
-import Data.Function (on)
-import Data.List (sortBy)
-import Data.Map as M
-import Data.Monoid
-import Data.Set as S
-import qualified Data.Text as T
-import Data.Text.Lens
-import Data.Time
-import Development.Shake
-import Development.Shake.Classes
-import Development.Shake.FilePath
-import GHC.Generics (Generic)
-import Slick
+import           Control.Concurrent
+import           Control.Lens
+import           Control.Monad              (forM_, forever, liftM, mzero, void,
+                                             when)
+import           Data.Aeson                 as A
+import           Data.Aeson.Lens
+import           Data.Either.Utils
+import           Data.Function              (on)
+import           Data.List                  as L (intercalate, intersperse,
+                                                  sortBy, (\\))
+import qualified Data.Map                   as M
+import           Data.Monoid
+import qualified Data.Set                   as S
+import qualified Data.Text                  as T
+import           Data.Text.Lens
+import           Data.Time
+import           Development.Shake
+import           Development.Shake.Classes
+import           Development.Shake.FilePath
+import           Development.Shake.Util     (shakeArgsPruneWith)
+import           GHC.Generics               (Generic)
+import           Slick
+import           Slick.Build
+import           Slick.Serve
+import           System.Console.GetOpt
+import           System.Directory
+import           System.Directory.Extra     (listFilesRecursive, removeFile)
+import           System.Environment
+
+import           Builder
+
+--------------------------------------------------------------------------------
+
+-- | List our custom keys that we need to manage additional behaviour
+--
+data Flags =
+    Preview     -- ^ key to run warp server
+--  | OtherFlag -- ^ run builder from clean slate
```

Comment:
Is this meant to be commented out?

Reply:
It's for novice users, to mention that one can multiple additional files. For educational purposes.